assistants/models.tsp

```diff
@@ -52,6 +119,27 @@ model CreateAssistantRequest {
    */
   @extension("x-oaiTypeLabel", "map")
   metadata?: Record<string> | null;
+
+  /**
+   * What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output
+   * more random, while lower values like 0.2 will make it more focused and deterministic.
+   */
+  @minValue(0)
+  @maxValue(2)
+  temperature?: float64 | null = 1;
```

Comment:
Do we want this to be float64 or just float?

![image](https://github.com/joseharriaga/openai-in-typespec/assets/21109913/0841b2ca-87d0-47c9-86cb-d61a9247d4ad)

Reply:
Good call, almost certainly just `float`!

Reply:
As I mentioned in another comment, it looks like the generator isn't fully friendly with `float`; here's a look at what happens with each thing:

```csharp
        /// <summary> Gets the test non nullable float. </summary>
        public object TestNonNullableFloat { get; }
        /// <summary> Gets the test nullable float. </summary>
        public object TestNullableFloat { get; }
        /// <summary> Gets the test non nullable float 32. </summary>
        public float TestNonNullableFloat32 { get; }
        /// <summary> Gets the test nullable float 32. </summary>
        public float? TestNullableFloat32 { get; }
        /// <summary> Gets the test non nullable float 64. </summary>
        public double TestNonNullableFloat64 { get; }
        /// <summary> Gets the test nullable float 64. </summary>
        public double? TestNullableFloat64 { get; }
```

We should discuss exactly what we want for these OpenAPI `number` types, but for now I'll move them to `float32` -- it's a very easy adjustment.

`float` seems to work fine in `@encode`, however, so I'll keep it in those decorators.